README.md

```diff
@@ -25,7 +25,56 @@ with API token or Jenkins CLI access but can also appear with features
 such as the [Authorize Project
 plugin](http://plugins.jenkins.io/authorize-project/)
 or other plugins that require details of user permissions or group
-membership outside of a user's web session)
+membership outside of a user's web session).
+
+### Compatibility Notes
+
+Various LDAP servers use different operational attributes to make decisions on
+and expose configurations of concepts such as disabling an account, locking an
+account, and specifying a time interval the account is valid for. These policies
+are normally enforced by the LDAP server itself when performing user authentication.
+Once a user has configured API tokens or SSH keys, these parallel authentication
+mechanisms can only make limited policy decisions based on checking for known user
+attributes without requiring the user to re-enter their password. These systems include
+[**slapo-ppolicy**(5)](https://linux.die.net/man/5/slapo-ppolicy) from OpenLDAP,
+Active Directory Application Mode (ADAM), classic Active Directory, and eDirectory LDAP.
+The support of these features is entirely dependent upon the LDAP server implementation
+properly exposing these operational attributes which is dependent on the exact version
+and distribution of the LDAP server in use.
+
+#### Administratively Disabled Accounts
+
+Accounts that are disabled by administrators typically expose an operational attribute
+to indicate such. The following attributes are all supported:
+
+* `pwdAccountLockedTime` value of `000001010000Z`: common LDAP attribute using password policy overlay;
+* `msDS-UserAccountDisabled` value of `TRUE`: modern [Active Directory attribute](https://docs.microsoft.com/en-us/windows/win32/adschema/a-msds-useraccountdisabled);
+* `userAccountControl` (UAC) or `msDS-User-Account-Control-Computed` with bit flag of `ADS_UF_ACCOUNTDISABLE` (0x2) present; and
+* `loginDisabled` value of `TRUE`: eDirectory attribute.
+
+#### Expired Accounts
```

Comment:
I really question using any time based calculations here. There are so many things that can go wrong when evaluating or comparing time. There is the question of interpreting the timestamps correctly. More significantly, there is a high likelihood of time zone differences or clock skew. If the LDAP server exists in a different time zone than the Jenkins controller, it may apply restrictions during the wrong times.

This section is rife with possibilities for being wrong and confusing things more.

Reply:
Timestamps have timezones specified. It's how LDAP GeneralizedTime values are stored. Are you suggesting we update the docs here or not support it at all?

Reply:
Added another sentence below clarifying the timezones.

Reply:
Good point. I'd still be concerned about clock skew, but that's probably not too big of a deal if we document it. I'm cautious about doing time-based comparisons based on data from another server but maybe it won't be too bad.